bindings/core/src/method_handler/client.rs

```diff
@@ -316,9 +319,12 @@ pub(crate) async fn call_client_method_internal(client: &Client, method: ClientM
             Response::BlockMetadata(client.get_block_metadata(&block_id).await?)
         }
         ClientMethod::GetBlockRaw { block_id } => Response::BlockRaw(client.get_block_raw(&block_id).await?),
-        ClientMethod::GetOutput { output_id } => {
-            Response::OutputWithMetadataResponse(client.get_output(&output_id).await?)
-        }
+        ClientMethod::GetOutput { output_id } => Response::OutputWithMetadataResponse(
+            client
+                .get_output(&output_id)
+                .await
+                .map(|o| OutputWithMetadataResponse::from(&o))?,
```

Comment:
Does this work?
```suggestion
                .map(OutputWithMetadataResponse::from)?,
```
Also regarding
>  should message_interface::Response be a DTO or non-DTO?

Yes, for the bindings we always want to use DTOs